File: examples/c++/collision.cxx
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2015 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/
#include <iostream>
#include <signal.h>

#include "collision.hpp"

using namespace std;

int shouldRun = true;

void
sig_handler(int signo)
{
    if (signo == SIGINT)
        shouldRun = false;
}

int
main(int argc, char** argv)
{
    signal(SIGINT, sig_handler);

    //! [Interesting]
    // The was tested with the Collision Sensor
    // Instantiate a Collision on digital pin D2
    upm::Collision collision(2);

    bool collisionState = false;
    cout << "No collision" << endl;
    while (shouldRun) {
        if (collision.isColliding() && !collisionState) {
            cout << "Collision!" << endl;
            collisionState = true;
        } else if (collisionState) {
            cout << "No collision" << endl;
            collisionState = false;
        }
    }

    //! [Interesting]
    cout << "Exiting" << endl;

    return 0;
}

File: examples/javascript/collision.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2015 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var collision_lib = require("jsupm_collision");

// The was tested with the Collision Sensor
// Instantiate a Collision on digital pin D2
var collision_obj = new collision_lib.Collision(2);

var collisionState = false;
console.log("No collision");

// Having an infinate loop prevents nodeJS from catching Cntl-C
// We need to catch Cntl-C to clean up memory
// Instead, we check the collision sensor every millisecond
var myInterval = setInterval(function()
{
	if (collision_obj.isColliding() && !collisionState)
	{
		console.log("Collision!");
		collisionState = true;
	}
	else if (!collision_obj.isColliding() && collisionState)
	{
		console.log("No collision");
		collisionState = false;
	}
}, 1);

// When exiting: clear interval, clean up memory, and print message
process.on('SIGINT', function()
{
	clearInterval(myInterval);
	collision_obj = null;
	collision_lib.cleanUp();
	collision_lib = null;
	console.log("Exiting...");
	process.exit(0);
});
